src/components/RepoForm/RepoForm.js

```diff
@@ -5,9 +5,22 @@ export default class RepoForm extends Component {
   static propTypes = {
     onSubmit: PropTypes.func.isRequired,
     error: PropTypes.object,
+    invalidFields: PropTypes.bool,
     loading: PropTypes.bool
   }
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      invalidFields: true
+    };
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    nextState.invalidFields = !(nextState.username && nextState.password && nextState.description && nextState.language);
```

Comment:
In React you're not suppose to mutate the states (Watch the videos on Redux tutorial about mutating). You should use the method this.setState({...state, invalidFields: <variable> }). Add this to handleChange where it checks the fields instead of adding it into componentWillUpdate.